pkg/lib/machines.js

```diff
@@ -554,15 +555,57 @@
                     state(host, "connecting", null);
             }
 
+            function load_manifest_plugins(packaged_manifests) {
+                if (manifest_plugins[host]) {
+                    manifest_plugins[host].forEach(function (pl) {
+                        pl.close();
+                        $(pl).off();
+                    });
+                }
+                manifest_plugins[host] = [ ];
+
+                function update_manifests() {
+                    var all_manifests = { };
+                    $.extend(true, all_manifests, packaged_manifests);
+                    manifest_plugins[host].forEach(function (pl) {
+                        if (pl.manifests)
+                            $.extend(true, all_manifests, pl.manifests);
+                    });
+                    machines.overlay(host, { manifests: all_manifests });
+                }
+
+                // XXX - This loads and evals all plugins in one go.
+                // Instead, we might want to eval them one by one so
+                // that a crash in one plugin does not disable all
+                // subsequent plugins.
+
+                $.ajax({ url: url + "/*/manifestplugin.js", dataType: "text", cache: true})
```

Comment:
> This is change of a fundamental principle that seems to need further discussion.

Yes.  I wasn't aware we have a plugin iframe...  In this case we would need one per host so that we can keep plugins from different machines separate.  I'll look around for the iframe...